Msix installer broken after upgrade to latest uno/uno.sdk/.net 8

### Current behavior

We previously created msix installers for our uno windows app using the below command line and everything worked fine

`msbuild.exe "path/to/uno.sln" /p:configuration="Release" /p:VisualStudioVersion="17.0" /p:AppxBundlePlatforms=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=SideLoadOnly /p:AppxPackageSigningEnabled=fasle /p:AppxPackageDir="C:\Code\app\AppxPackages" `

After upgrading from .net 7 to .net 8 as well as latest uno and uno.sdk the same command line still produces the msix but the app crashes on startup. The only error I can get is from the event log

```
Faulting application name: UnoApp1.Windows.exe, version: 1.0.0.0, time stamp: 0x65a80000
Faulting module name: KERNELBASE.dll, version: 10.0.19041.3930, time stamp: 0x9141f979
Exception code: 0xc000027b
Fault offset: 0x000000000012db22
Faulting process ID: 0x6104
Faulting application start time: 0x01da6023a0051583
Faulting application path: C:\Program Files\WindowsApps\UnoApp1_1.0.0.0_x64__h21bem17p9w1p\UnoApp1.Windows.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report ID: 1ffb248d-9dea-4b31-9e09-90dba8267c53
Faulting package full name: UnoApp1_1.0.0.0_x64__h21bem17p9w1p
Faulting package-relative application ID: App
```

Crucially, on the same machine, if instead of running the msbuild command line I open Visual studio and right click->Package and Publish the created msix works fine and the app runs. This tells me it is not to do with the setup of the machine but the command line used to create the installer.


### Expected behavior

Providing the correct command(s) needed to create a functioning msix installer

### How to reproduce it (as minimally and precisely as possible)

I recreated this using a uno windows application created from latest templates

then running the command

`msbuild.exe "path/to/uno.sln" /p:configuration="Release" /p:VisualStudioVersion="17.0" /p:AppxBundlePlatforms=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundle=Never /p:UapAppxPackageBuildMode=SideLoadOnly /p:AppxPackageSigningEnabled=fasle /p:AppxPackageDir="C:\Code\app\AppxPackages" `

Installing the produced msix

Run the app, crashes on startup

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

Windows

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

Thankyou @jeromelaban 

This setting has fixed our windows installer. The app now runs without crashing after install.

We did have CI processes that use `dotnet build` that started failing because of this setting. I had to switch our CI builds to use `msbuild `instead of `dotnet build` as a workaround.

dotnet build fails with 
`error MSB4062: The "Microsoft.Build.Packaging.Pri.Tasks.ExpandPriContent" task could not be loaded from the assembly`

which is an error ive seen in the past and workaround using `EnablePreviewMsixTooling`

It would be nice to properly understand what these various settings do but i cant find anything from MS on them. Can you add anything esp around the dotnet build/msbuild discrepancy

--- a/src/Uno.Sdk/targets/Uno.Common.WinAppSdk.targets
+++ b/src/Uno.Sdk/targets/Uno.Common.WinAppSdk.targets
@@ -4,12 +4,17 @@
 		<TargetPlatformMinVersion Condition=" $(TargetPlatformMinVersion) == '' ">10.0.18362.0</TargetPlatformMinVersion>
 		<SupportedOSPlatformVersion Condition=" $(SupportedOSPlatformVersion) == '' ">$(TargetPlatformMinVersion)</SupportedOSPlatformVersion>
 		<RuntimeIdentifiers Condition=" $(RuntimeIdentifiers) == '' ">win-x86;win-x64;win-arm64</RuntimeIdentifiers>
-		<EnableCoreMrtTooling Condition=" $(EnableCoreMrtTooling) == '' AND '$(BuildingInsideVisualStudio)' != 'true' ">false</EnableCoreMrtTooling>
 
 		<!-- Set EnableMsixTooling to true only for executables - setting this on class libraries prevents assets from being correctly copied to windows target -->
 		<EnableMsixTooling Condition=" $(EnableMsixTooling) == '' AND ('$(OutputType)' == 'WinExe' OR '$(OutputType)' == 'Exe') ">true</EnableMsixTooling>
 
 		<EnableWindowsTargeting Condition=" $(EnableWindowsTargeting) == '' ">true</EnableWindowsTargeting>
+	</PropertyGroup>
+
+	<PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
+		<!-- Workaround Uno Issue 15492 - This must be set to true for the Library so that the generated files use the Library name in the path -->
+		<EnableCoreMrtTooling Condition=" $(EnableCoreMrtTooling) == '' AND $(OutputType) != 'WinExe' AND $(OutputType) != 'Exe' ">true</EnableCoreMrtTooling>
+		<EnableCoreMrtTooling Condition=" $(EnableCoreMrtTooling) == '' AND '$(MSBuildRuntimeType)' == 'Core' ">false</EnableCoreMrtTooling>
 	</PropertyGroup>
 
 	<PropertyGroup Condition="$(SingleProject)!='true' and $(OutputType)=='WinExe'">

